gqcp/tests/QCMethod/HF/GHF/QCMethod_GHF_stability_test.cpp

```diff
@@ -0,0 +1,158 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#define BOOST_TEST_MODULE "QCMethod_GHF_stability_test"
+
+#include <boost/test/unit_test.hpp>
+
+#include "Basis/SpinorBasis/GSpinorBasis.hpp"
+#include "Operator/SecondQuantized/SQHamiltonian.hpp"
+#include "QCMethod/HF/GHF/GHF.hpp"
+#include "QCMethod/HF/GHF/GHFSCFSolver.hpp"
+#include "QCModel/HF/StabilityMatrices/GHFStabilityMatrices.hpp"
+
+/**
+ *  Starting from a core guess, the GHF SCF algorithm finds a solution that should be both internally and externally unstable for the given system.
+ *  This test checks whether the stability checks confirm this.
+ * 
+ *  The system of interest is a H3-triangle, 1 bohr apart and the reference implementation was done by @xdvriend.
+ */
+BOOST_AUTO_TEST_CASE(H3_stability_test_1) {
+
+    // Set up a general spinor basis to obtain a spin-blocked second-quantized molecular Hamiltonian.
+    const auto molecule = GQCP::Molecule::HRingFromDistance(3, 1.0);  // H3-triangle, 1 bohr apart
+    const auto N = molecule.numberOfElectrons();
+
+    const GQCP::GSpinorBasis<double, GQCP::GTOShell> g_spinor_basis {molecule, "STO-3G"};
+    const auto S = g_spinor_basis.overlap().parameters();
+
+    const auto sq_hamiltonian = GQCP::GSQHamiltonian<double>::Molecular(g_spinor_basis, molecule);
+
+    auto environment = GQCP::GHFSCFEnvironment<double>::WithCoreGuess(N, sq_hamiltonian, S);
```

Comment:
Maybe we could put this 2 lines below? It fits better in that 'section' of the test.